libsolidity/ast/ASTJsonExporter.cpp

```diff
@@ -468,6 +468,10 @@ bool ASTJsonExporter::visit(FunctionDefinition const& _node)
 		attributes.emplace_back("functionSelector", _node.externalIdentifierHex());
 	if (!_node.annotation().baseFunctions.empty())
 		attributes.emplace_back(make_pair("baseFunctions", getContainerIds(_node.annotation().baseFunctions, true)));
+
+	if (_node.annotation().internalFunctionID.set())
```

Comment:
This would need to both check that the ``SetOnce`` is set *and* that the ``optional`` has a value, wouldn't it? I.e.
```suggestion
	if (_node.annotation().internalFunctionID.set() && _node.annotation().internalFunctionID->has_value())
```

Reply:
Also in a few other places - but you probably *only* even set it, if it has a value? Then it should probably just be ``util::SetOnce<uint64_t>``, shouldn't it?

Reply:
Yeah, looks like it's the latter - so we can just drop the `optional` part.

Reply:
Yeah, [set()](https://github.com/ethereum/solidity/blob/960889f532aad8f5de0fbb3bf0822f349300fb5f/libsolutil/SetOnce.h#L83) already check if it `has_value()`. Maybe we should rename it to `is_set()` or something.

Reply:
Just to be clear: The point is that a ``SetOnce<std::optional<T>>`` like we have now has *two* layers of optionals and we only need one, i.e. this should still be changed to ``SetOnce<uint64_t>`` instead of ``SetOnce<optional<uint64_t>>``.